### PR TITLE
[DSM] PEPPER-139 Adding new access role for viewing clinical orders

### PR DESCRIPTION
**Pull request by pegahtah**
I have added the new role as the ones that allows the user to view the sequencing tab and the cliniical orders page. They still can not submit orders

### CODE REVIEW

File: ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.html
Comment (pegahtah) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.html:82]: There is a guard in the backend that doesn't let the orders go through without proper "submit orders" access role, but I am also removing the buttons from the view if they  only have the view role

Comment (denniscunningham) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.html:82]: Just so I understand, this is a 'drive-by' fix... not required for this task, but something you noticed that was incorrect?

File: ddp-workspace/projects/ddp-dsm-ui/src/app/navigation/navigation.component.html
Comment (denniscunningham) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/navigation/navigation.component.html:20]: How does 'allowedToDoOrderSequencing()'  fit into this feature?

Comment (otchet-broad) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/navigation/navigation.component.html:20]: Is viewSeqOrderStatus a property or a function?

Comment (pegahtah) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/navigation/navigation.component.html:20]: I created a test user with only this new permission and participant list permission , and realized that they won't see the `Samples` menu since it only checked for samples and kit shipping access, while people with either of `allowedToDoOrderSequencing` or `viewSeqOrderStatus` should be able to see the `Clinical Orders` menu point in this drop down. That's why I corrected this.

File: ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.html
Comment (pegahtah) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.html:82]: It is required in this task that the user with only this new permission, should not be able to submit an order. There is a backend check for this, but I also removed the submit button

File: ddp-workspace/projects/ddp-dsm-ui/src/app/navigation/navigation.component.html
Comment (pegahtah) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/navigation/navigation.component.html:20]: @otchet-broad `viewSeqOrderStatus` is a get function

Comment (denniscunningham) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/navigation/navigation.component.html:20]: I assumed the issue here was the lack of parens after 'viewSeqOrderStatus'?

Comment (pegahtah) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/navigation/navigation.component.html:20]: Thank you for catching this! It was a naming issue where both the variable and the function had the same name. I fixed it now

File: ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.html
Comment (pegahtah) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.html:6]: I removed the selection check boxes because they are not useful anymore in a view only permission

File: ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.html
Comment (pegahtah) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.html]: I also removed the date submission, but we are waiting on the final decision on that

**Review by denniscunningham — APPROVED** [review]
LGTM, but you may want approval from others.

**Review by otchet-broad — APPROVED** [review]
Thanks for diving in and answering that question @pegahtah!

File: ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts
Comment (denniscunningham) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts]: Do you want a return type?
Is the indenting a bit off, or is this just a GIthub view thing?

File: ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts
Comment (otchet-broad) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts]: I can't remember with the version of underscore we're using, but I thought _.isEmpty returns true for undefined and null as well.  Is this similar to something else in our code?  If so, have a pointer?

Comment (otchet-broad) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts]: What's the type of sample.collectionDate?  From the [underscore documentation](https://underscorejs.org/#isEmpty):
isEmpty_.isEmpty(collection) [source](https://underscorejs.org/docs/modules/isEmpty.html)
Returns true if collection has no elements. For **strings** and array-like objects _.isEmpty checks if the length property is 0. For other objects, it returns true if the object has no enumerable own-properties. Note that primitive numbers, booleans and symbols are always empty by this definition.

File: ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts
Comment (otchet-broad) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts]: +1

File: ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts
Comment (otchet-broad) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts]: It looks like this is a string.  If that's the case, this is fine as is, but likely still could be reduced to just _.isEmpty(sample.collectionDate) because these test pass:

      expect(_.isEmpty(null)).toBe(true);
      expect(_.isEmpty(undefined)).toBe(true);

File: ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts
Comment (pegahtah) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts]: fixed, thank you :)

File: ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts
Comment (pegahtah) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts]: it is a string, that's why I kept it as is

**Review by denniscunningham — APPROVED** [review]

**Review by otchet-broad — APPROVED** [review]

**Pull request merged**
